nixos/gonic: fix writeability of paths from config

katexochen opened the pull request:
Not sure this is even possible with DynamicUser. Even if we BindPath the directories the dynamic service user isn't allowed to write them, they are root owned. 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

katexochen committed:
nixos/gonic: add package option

Signed-off-by: Paul Meyer <katexochen0@gmail.com>

katexochen committed:
nixos/gonic: ensure paths from config are writable

Signed-off-by: Paul Meyer <katexochen0@gmail.com>

autrimpo reviewed (changes requested):
Thank you! I think the best thing to do is to get rid of the DynamicUsers and run it as root, it's causing a lot of permission issues for people. Running as root should make whatever user-specified location "just work". I checked using the nixos test and since we set `RootDirectory = "/run/gonic";` and bind mount the user directories, there's not much extra the service can access.

Not sure about the cache path assertion in this case, that should not be necessary now, no? Or is there something besides the possible permission issues that requires it?

katexochen committed:
nixos/gonic: remove DynamicUser

It's not possible to let a user hand in paths that gonic should
write to if DynamicUser is used, remove it. The service should
still be sufficiently sandboxed as using chroot and only binding
paths selectively.

Signed-off-by: Paul Meyer <katexochen0@gmail.com>

katexochen committed:
nixosTests.gonic: add maintainers

Signed-off-by: Paul Meyer <katexochen0@gmail.com>
